### PR TITLE
build: publish build artifacts to branches

### LINKED ISSUE #13126: change publish-build-artifacts.sh to publish non-master branch builds to a custom branch
`publish-build-artifacts.sh` currently publishes build artifacts from any `angular/angular` (upstream) travis builds to `angular/core-builds` and friends.

It always uses the master branch in these `-builds` repos, even for builds that originate from patch branches.

The docs team needs to be able to pull in master and 2.2.x (soon to be 2.3.x) builds separately which is currently not possible.

We should modify the `publish-build-artifacts.sh` script to publish into a branch that matches the current branch (if such branch doesn't exist in angular/core-builds and friends, this branch should be created automatically).

// @wardbell

### PR DESCRIPTION
**Pull request by filipesilva**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
See #13126

**What is the new behavior?**
The currently checked out branch will be used as the destination branch in the snapshot repos. If the branch does not exist, it will be created.

Build versions are now `${LATEST_TAG}+${SHORT_SHA}` instead of `2.0.0-${SHORT_SHA}` (e.g. `4.0.0-beta.0+31d3eba` instead of `2.0.0-31d3eba`).

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Fix #13126
/cc @IgorMinar

### CODE REVIEW

**Comment by filipesilva** [comment]
I'm not aware of a test for the `publish-build-artifacts.sh` script. This is what I did to test my solution:
```
git clone https://github.com/angular/angular.git
cd angular
# make mock package content to test
mkdir -p dist/packages-dist/mock/
echo "mock" > dist/packages-dist/mock/file.txt
mkdir -p dist/tools/@angular/tsc-wrapped/
echo "mock" > dist/tools/@angular/tsc-wrapped/file.txt
# make new repos on github:
# https://github.com/filipesilva/mock-builds
# https://github.com/filipesilva/tsc-wrapped-builds
./scripts/publish/publish-build-artifacts.sh filipesilva
# repos have now the latest commit from master
git co -b test-branch
./scripts/publish/publish-build-artifacts.sh filipesilva
# test-branch was created
```

File: scripts/publish/publish-build-artifacts.sh [hunk 1]
```diff
@@ -23,17 +23,16 @@ function publishRepo {
     cd $REPO_DIR && \
     git init && \
     git remote add origin $REPO_URL && \
-    git fetch origin master --depth=1 && \
-    git checkout origin/master && \
-    git checkout -b master
+    git fetch origin --depth=1 && \
+    git checkout -b "${BRANCH}"
```
Comment (IgorMinar) [inline scripts/publish/publish-build-artifacts.sh]: won't this fail if the branch already exists?

Comment (filipesilva) [inline scripts/publish/publish-build-artifacts.sh]: Worse, it will replace the branch if it exists instead of adding to it. Will fix.

**Comment by IgorMinar** [comment]
thanks @filipesilva !!!

**Pull request merged**